### PR TITLE
_3mux: 0.2.0 -> 1.0.1

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/aaronjanse/3mux/releases/tag/v1.0.1

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aaronjanse** [comment]
What's the best way to remove the `fuzz` binary from `$OUT/bin`? Maybe a simple post-install `rm` command?

**Comment by aaronjanse** [comment]
Would you mind adding the following, based on the way [gotools removes an unneeded module](https://github.com/NixOS/nixpkgs/blob/2bba291913d19ad31e35988783f4e5f2089b9ddb/pkgs/development/tools/gotools/default.nix#L14-L22)?

```
  postPatch = ''
    rm -rf fuzz
  '';
```

**Comment by filalex77** [comment]
@Mic92 I read through #88258, but I have a question about `vendorSha256`: is the actual SHA-256 *exactly the same* as `modSha256` (e.g. it's only a rename), or is it generated differently?

**Comment by Mic92** [comment]
> @Mic92 I read through #88258, but I have a question about `vendorSha256`: is the actual SHA-256 _exactly the same_ as `modSha256` (e.g. it's only a rename), or is it generated differently?

It is generated differently hence the rename.

File: pkgs/tools/misc/3mux/default.nix [hunk 1]
```diff
@@ -21,4 +23,4 @@ buildGoModule rec {
     # TODO: fix modules build on darwin
     broken = stdenv.isDarwin;
```
Comment (aaronjanse) [inline pkgs/tools/misc/3mux/default.nix:24]: For some reason, `3mux` builds on darwin but cross-compiling to darwin from linux doesn't work. I think this is `3mux`'s fault somehow

Comment (filalex77) [reply pkgs/tools/misc/3mux/default.nix]: @aaronjanse Thanks for letting me know.

@GrahamcOfBorg build _3mux

**Comment by filalex77** [comment]
@GrahamcOfBorg build _3mux

**Comment by aaronjanse** [comment]
I think I figured it out: `3mux` cross-compiles to darwin if `CGO_ENABLED = 1`

**Comment by aaronjanse** [comment]
I don't have a darwin setup to test this on, but other than that, this LGTM

**Review by aaronjanse — APPROVED** [review]

**Review by aaronjanse — CHANGES_REQUESTED** [review]
Oh wait. I think CGO_ENABLED should be enabled for all OSes?

**Comment by Mic92** [comment]
@GrahamcOfBorg build _3mux

File: pkgs/tools/misc/3mux/default.nix [hunk 2]
```diff
@@ -2,23 +2,27 @@
...
+
+  excludedPackages = [ "fuzz" ];
 
   meta = with stdenv.lib; {
     description = "Terminal multiplexer inspired by i3";
     homepage = "https://github.com/aaronjanse/3mux";
     license = licenses.mit;
     maintainers = with maintainers; [ aaronjanse filalex77 ];
-    # TODO: fix modules build on darwin
-    broken = stdenv.isDarwin;
```
Comment (Mic92) [inline pkgs/tools/misc/3mux/default.nix:22]: Looks like this is still broken on macOS itself.

**Review by bhipple — CHANGES_REQUESTED** [review]
CI failing

**Comment by Lassulus** [comment]
meanwhile: this was bumped to 0.3.0

**Comment by aaronjanse** [comment]
See https://github.com/NixOS/nixpkgs/pull/116272 for an update to v1.1.0. Thank you @Br1ght0ne!

**Pull request closed without merging**